frontend/src/service/DoctorApiService.js

```diff
@@ -3,3 +3,13 @@ import axios from 'axios';
 export function submitDoctor(newDoctor) {
 	return axios.post('/api/doctor', newDoctor).then((response) => response.data);
 }
+
+export function submitAppointment(newDoctor) {
+	return axios
+		.put('/api/doctor/appointment', newDoctor)
```

Comment:
technically not rest conform to my understanding. you're updating appointment attribute of the doctor, so it should be put(api/doctor/doctorId, newDoctor)
Maybe not so important after all...